contracts/schemes/ContributionRewardExt.sol

```diff
@@ -0,0 +1,454 @@
+        }
+
+        if (reputation > 0) {
+            require(
+            Controller(
+            avatar.owner()).mintReputation(uint(reputation), _proposal.beneficiary, address(avatar)));
+        } else if (reputation < 0) {
+            require(
+            Controller(
+            avatar.owner()).burnReputation(uint(reputation*(-1)), _proposal.beneficiary, address(avatar)));
+        }
+        if (reputation != 0) {
+            emit RedeemReputation(address(avatar), _proposalId, _proposal.beneficiary, reputation);
+        }
+    }
+
+    /**
+    * @dev RedeemNativeToken reward for proposal
+    * @param _proposalId the ID of the voting in the voting machine
+    * @return amount the redeemed nativeToken.
+    */
+    function redeemNativeToken(bytes32 _proposalId) public returns(uint256 amount) {
+
+        ContributionProposal memory _proposal = organizationProposals[_proposalId];
+        ContributionProposal storage proposal = organizationProposals[_proposalId];
+        require(proposal.executionTime != 0);
+        //set proposal rewards to zero to prevent reentrancy attack.
+        proposal.nativeTokenReward = 0;
+
+        if (_proposal.beneficiary == address(this)) {
+            if (_proposal.nativeTokenReward != 0) {
+                proposal.nativeTokenRewardLeft = _proposal.nativeTokenReward;
+            }
+        }
+        amount = _proposal.nativeTokenReward;
+        if (amount > 0) {
+            require(Controller(avatar.owner()).mintTokens(amount, _proposal.beneficiary, address(avatar)));
+            emit RedeemNativeToken(address(avatar), _proposalId, _proposal.beneficiary, amount);
+        }
+    }
+
+    /**
+    * @dev RedeemEther reward for proposal
+    * @param _proposalId the ID of the voting in the voting machine
+    * @return amount ether redeemed amount
+    */
+    function redeemEther(bytes32 _proposalId) public returns(uint256 amount) {
+        ContributionProposal memory _proposal = organizationProposals[_proposalId];
+        ContributionProposal storage proposal = organizationProposals[_proposalId];
+        require(proposal.executionTime != 0);
+        //set proposal rewards to zero to prevent reentrancy attack.
+        proposal.ethReward = 0;
+        if (_proposal.beneficiary == address(this)) {
+            if (_proposal.ethReward != 0) {
+                proposal.ethRewardLeft = _proposal.ethReward;
+            }
+        }
+        amount = _proposal.ethReward;
+
+        if (amount > 0) {
+            require(Controller(avatar.owner()).sendEther(amount, _proposal.beneficiary, avatar));
+            emit RedeemEther(address(avatar), _proposalId, _proposal.beneficiary, amount);
+        }
+    }
+
+    /**
+    * @dev RedeemNativeToken reward for proposal
+    * @param _proposalId the ID of the voting in the voting machine
+    * @return amount the external token redeemed amount
+    */
+    function redeemExternalToken(bytes32 _proposalId) public returns(uint256 amount) {
```

Comment:
wondering if we should be anticipating here the ability to reward multiple external tokens?

Reply:
we might add that later on.not in this phase.